docs/guides/app-dev/mempool-lanes.md

```diff
@@ -0,0 +1,163 @@
+# Mempool Lanes
+
+The Lanes feature allows applications to classify and prioritise transactions for providing Quality
+of Service guarantees to the mempool ([ADR-118](adr)). 
+
+This guide provides instructions, along with a set of best practices and rules of thumb to help
+setting up and using Lanes within your application. 
+
+## How to set up lanes in an application
+
+Application developers that want to make use of Lanes must pre-define a list of lane names and their
+priorities and then populate specific fields in `Info` and `CheckTx` responses. In contrast, if
+Lanes are not to be used by the application, no modifications to the code are necessary.
+
+We will explain with an example taken from our implementation of Lanes in the `kvstore` application
+(in `abci/example/kvstore/kvstore.go`).
+
+### Define lanes and their priorities
+
+First, the application should keep a list of lane IDs (of type `string`) and their priorities (of
+type `uint32`). In this example we store it as a map in the `Application` struct. And we define as a
+constant the lane ID used as default when assigning lanes to transactions.
+
+```go
+const defaultLane = "C"
+
+type Application struct {
+  ...
+  lanePriorities map[string]uint32
+}
+
+func NewApplication(...) *Application {
+  ...
+  return &Application{
+    ...
+    lanePriorities: map[string]uint32{
+      "A": 100,
+      "B": 50,
+      defaultLane: 10,
+      "D": 1,
+    },
+  }
+}
+```
+
+### Handling Info requests
+
+When a CometBFT node starts, it performs a handshake with the application by sending an `Info`
+request. This process allows CometBFT to retrieve essential data from the application to initialize
+and configure itself.
+
+Upon receiving an `Info` request, the application must reply with the lane IDs and
+their priorities in the `LanePriorities` field, and the default lane in the `DefaultLane` field. The
```

Comment:
For some reason, I can't add suggestions.

I would phrase that the `LanePriorities` field is a `map[string]uint32`.